tests/schedulers/test_arq.py

```diff
@@ -0,0 +1,142 @@
+    myworker = arq_worker(config, burst=True)
+    yield myworker
+    await myworker.close()
+
+
+@pytest.fixture
+async def scheduler(config: ArqSchedulerConfig) -> AsyncGenerator[ArqScheduler, None]:
+    myscheduler = ArqScheduler(config)
+    yield myscheduler
+    await myscheduler.close()
+
+
+@pytest.fixture
+def job_dir(tmp_path: Path) -> Path:
+    return tmp_path
+
+
+@pytest.mark.anyio
+async def test_state_for_unknown_job_id(scheduler: ArqScheduler) -> None:
+    with pytest.raises(KeyError):
+        await scheduler.state("unknownjobid")
+
+
+def test_eq_good(scheduler: ArqScheduler, redis_dsn: RedisDsn) -> None:
+    same = ArqScheduler(ArqSchedulerConfig(redis_dsn=redis_dsn))
+    assert scheduler == same
+
+
+def test_eq_diff_class(scheduler: ArqScheduler) -> None:
+    diff = "somethingelse"
+    assert scheduler != diff
+
+
+def test_eq_diff_config(scheduler: ArqScheduler) -> None:
+    diff = ArqScheduler(
+        ArqSchedulerConfig(redis_dsn="redis://somehost", queue="somequeue"),
+    )
+    assert scheduler != diff
+
+
+@pytest.mark.anyio
+async def test_repr(scheduler: ArqScheduler, redis_dsn: RedisDsn) -> None:
+    redis = f"redis_dsn=RedisDsn('{redis_dsn}'"
+    config = f"ArqSchedulerConfig(type='arq', {redis}, ), queue='arq:queue')"
+    expected = f"ArqScheduler(config={config})"
+    assert repr(scheduler) == expected
+
+
+@pytest.fixture
+async def submitted_job_id(scheduler: ArqScheduler, job_dir: Path) -> str:
+    description = prepare_input(job_dir)
+    return await scheduler.submit(description)
+
+
+@pytest.mark.anyio
+async def test_submitted_job_has_queued_state(
+    scheduler: ArqScheduler,
+    submitted_job_id: str,
+) -> None:
+    state = await scheduler.state(submitted_job_id)
+    assert state == "queued"
+
+
+@pytest.fixture
+async def completed_job_id(submitted_job_id: str, worker: Worker) -> str:
+    await worker.main()
+    return submitted_job_id
+
+
+class TestCompletedJob:
+    @pytest.mark.anyio
+    async def test_has_ok_state(
+        self,
+        scheduler: ArqScheduler,
+        completed_job_id: str,
+    ) -> None:
+        state = await scheduler.state(completed_job_id)
+        assert state == "ok"
+
+    def test_has_output(self, completed_job_id: str, job_dir: Path) -> None:
+        assert_output(job_dir)
+
+
+@pytest.mark.anyio
+async def test_cancelling_queued_job(
+    scheduler: ArqScheduler,
+    tmp_path: Path,
+    config: ArqSchedulerConfig,
+) -> None:
+    job_dir = tmp_path
+    description = prepare_input(job_dir)
+    submitted_job_id = await scheduler.submit(description)
+    queued_state = await scheduler.state(submitted_job_id)
+    assert queued_state == "queued"
+
+    await scheduler.cancel(submitted_job_id)
+
+    # Run worker as it does aborting
+    worker = arq_worker(config, burst=True)
+    await worker.main()
+
+    state = await scheduler.state(submitted_job_id)
+    assert state == "error"
```

Comment:
Arq does not have any according to https://github.com/samuelcolvin/arq/blob/main/arq/jobs.py#L25-L39 
Do you mean a state like cancelling, similar to pending, running? I am not sure if we need protection against cancelling a job that is already being cancelled.

At the moment jobs that error out due to full disk, permission, service down, etc. or jobs that are cancelled  explicitly by a user both set to 'error' state. 

It could be useful to distinguish the two.
Do you mind creating an issue for a adding a cancelled state?

Reply:
#53